### PR TITLE
Add BOM definition to the OTel starter

### PR DESCRIPTION
**Pull request by jeanbisutti**
This PR explains how to configure the `opentelemetry-bom` and `opentelemetry-instrumentation-bom-alpha` BOMs for the Spring Boot starter page.

For the  Java manual instrumentation page, @PerfectSlayer has already created a PR: https://github.com/open-telemetry/opentelemetry.io/pull/3656

### CODE REVIEW

**Comment by jeanbisutti** [comment]
Please note that the `opentelemetry-instrumentation-bom-alpha` BOM is only required for the Spring Boot page, not the the Java manual instrumentation page.

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md:57]: The Spring Boot documentation is using bom: https://docs.spring.io/spring-boot/docs/current/gradle-plugin/reference/htmlsingle/

The Maven (https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html) and Gradle (https://docs.gradle.org/current/userguide/platforms.html) documentations are using BOM.

BOM means "Bill Of Materials".

So, I would say BOM.

**Review by theletterf — APPROVED** [review]
Thanks!

**Comment by jeanbisutti** [comment]
@theletterf I have added 2 commits:
*  [Fix missing words](https://github.com/open-telemetry/opentelemetry.io/pull/3672/commits/1908228be4afabbd50d0332e0b7923834f28ce7f)
*  [Add more explanations](https://github.com/open-telemetry/opentelemetry.io/pull/3672/commits/587c5d6e84ebc798bd05a9c0d803044c03421551) The configuration has confused several OTel user

@theletterf Could you please review again? Thank you! 🙏

**Review by theletterf — APPROVED** [review]

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:26]:
```suggestion
### Bill of Materials
```

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:29]:
```suggestion
Import the `opentelemetry-bom` and `opentelemetry-instrumentation-bom-alpha` Bill of Materials (BOMs) before using the OpenTelemetry Starter.
```

Can you provide some context to the reader why they may want to, should or have to include this?

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:56]:
```suggestion
You can use the Gradle’s native BOM support:
```

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:95]:
```suggestion
{{% alert title="Note" color="info" %}} 

Be careful not to mix up the different
ways of configuring things with Gradle. For example, don't use
`implementation(platform(SpringBootPlugin.BOM_COORDINATES))` with the
`io.spring.dependency-management` plugin.

{{% /alert %}}
```

If you add additional new lines the alert tags remain in their own lines

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:97]:
```suggestion
### OpenTelemetry Starter dependency
```

**Review by svrnm — CHANGES_REQUESTED** [review]

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md:29]: I have added explanations: 3080196cec7cd23e209ac07a732209cb7e6b7eca

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (trask) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:32]: (maybe?)
```suggestion
The `opentelemetry-bom` BOM will align the OpenTelemetry versions
used by the OpenTelemetry starter and its transitive dependencies.
```

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (trask) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:102]:
```suggestion
`implementation(platform("io.opentelemetry:opentelemetry-bom:{{% param vers.otel %}}"))` with the
```

**Comment by trask** [comment]
thx! I think this will resolve #3613

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (svrnm) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md:29]: I will take another look, thx

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (svrnm) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:29]: Some people might not know what a BOM is and why they need it. As of today this is the first reference to this topic, so adding a few words on what it is and why you want it/need it, would be helpful to the reader.

**Comment by svrnm** [comment]
@jeanbisutti there is no need to run fix : format or any other command after each commit, we can satisfy the Checks before merging this PR. This way the comments stay a little bit more clear. Thanks!

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md:29]: @svrnm A BOM is something usual in the Java world. [This Spring Boot documentation](https://docs.spring.io/spring-boot/docs/current/gradle-plugin/reference/htmlsingle/) just use "bom". I have added explanations about the interest of using BOMs with the OTel starter:

    The `opentelemetry-bom` BOM will allow you to align the OpenTelemetry versions
    of the OpenTelemetry dependencies and transitive dependencies.
    
    With the `opentelemetry-instrumentation-bom-alpha` BOM, you won't have to
    declare the dependency versions for additional instrumentations.

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md]: I have added a more generic formulation.

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md]: @svrnm I have added a link to the BOM definition: 3783e53408a1359c2681502ac8f7dfe17db34b63 Do you think that it's a good trade off to not re-explain a classical notion in the Java world but to offer the possibility to quickly have the definition for some junior Java developers?

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (trask) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:35]: tried to simplify a bit and make it more beginner friendly, not sure if I succeeded...

```suggestion
### Dependencies

Importing the `opentelemetry-bom` and `opentelemetry-instrumentation-bom-alpha` BOMs will ensure that the versions of your OpenTelemetry dependencies (and transitive dependencies) are aligned.
```

Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md]: I prefer this version. Thanks!

Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md]: @svrnm What do you think? Would it be good enough for merging?

Comment (svrnm) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md]: What's missing for me: the term BOM needs to be written out when used for the first time Bill Of Materials (BOM), and then some light reason what it is (in general terms) and why someone wants to have that, e.g.

```
A Bill of Materials (BOM) is a summary of the components used within a software artifact, which helps with compliance and security. By importing the ...
```

Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md]: @svrnm Could you please check the new version?

**Review by svrnm — APPROVED** [review]

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (svrnm) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md]: lgtm now. @trask any more comments?

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (trask) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:26]:
```suggestion
### Dependency management
```

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (trask) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:30]:
```suggestion
Importing the `opentelemetry-bom` and `opentelemetry-instrumentation-bom-alpha` BOMs when using the OpenTelemetry starter is important to ensure version alignment across all OpenTelemetry dependencies.
```

**Review by trask — APPROVED** [review]
last couple of thoughts

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md:26]: I would prefer to keep "Bill of Materials" because the next section is  "OpenTelemetry Starter dependency"

Comment (trask) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md:26]: my thought is that someone who doesn't know what Bill Of Materials are may skip over the section and go straight to "OpenTelemetry Starter dependency"

**Comment by trask** [comment]
@zeitlinger if you have a chance can you give this a review?

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md]: @zeitlinger Please check fb969bf2761b1fe82e59ed509aa610d1b5b79a16

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (jeanbisutti) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md:92]: I don't know for Dependabot. Renovate only supports the first mechanism: https://github.com/renovatebot/renovate/issues/4614#issuecomment-557488428. I am hesitant to add this information about Renovate because Renovate may also support the second mechanism in the future.

Comment (zeitlinger) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md:92]: ok

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (zeitlinger) [reply content/en/docs/instrumentation/java/automatic/spring-boot.md]: sounds good

**Comment by jeanbisutti** [comment]
Thank you for the review @zeitlinger 

cc @svrnm

**Comment by svrnm** [comment]
> Thank you for the review @zeitlinger

Thanks from me as well, @zeitlinger.

Both, let me know when you are through the review process from SIG Java perspective, and don't worry too much about the CI checks while doing so, I can help to fix them before merging.

**Comment by zeitlinger** [comment]
@svrnm good from my side

File: content/en/docs/instrumentation/java/automatic/spring-boot.md
Comment (trask) [inline content/en/docs/instrumentation/java/automatic/spring-boot.md:28]: (to make it sound more important as opposed to an optional thing. even if it is optional, we want to strongly encourage it)
```suggestion
A Bill of Material ([BOM](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#bill-of-materials-bom-poms)) ensures that versions of dependencies (including transitive ones) are aligned.
```

**Review by trask — APPROVED** [review]

**Comment by jeanbisutti** [comment]
@svrnm Trask has approved on behalf of java-approvers

**Comment by trask** [comment]
thx @jeanbisutti @zeitlinger @svrnm!

**Pull request merged**
